File: solutions/julia/nucleotide-count/1/nucleotide-count.jl
"""
    count_nucleotides(strand)

The count of each nucleotide within `strand` as a dictionary.

Invalid strands raise a `DomainError`.

"""
function count_nucleotides(strand)
    count = Dict('A' => 0, 'C' => 0, 'G' => 0, 'T' => 0)
    for c in strand
        if c in "GCAT"
            count[c] += 1
        else
            throw(DomainError(c))
        end
    end
    count
end

File: solutions/javascript/nucleotide-count/1/nucleotide-count.js
export function countNucleotides(strand) {
  const regex = /^[GATC]*$/g;
  if (strand.match(regex) === null) {
    throw new Error("Invalid nucleotide in strand");
  }
  const nucleotides = {A: 0, C: 0, G: 0, T: 0};
  for(const n of strand) {
    nucleotides[n]++;
  }
  return `${nucleotides.A} ${nucleotides.C} ${nucleotides.G} ${nucleotides.T}`;
}
